File: Closure/8 kyu - Multiply.clj
; https://www.codewars.com/kata/50654ddff44f800200000004/train/clojure
; My solution
(ns multiply.bug.fix)

(defn multiply [a b] (* a b))

; ...
(ns multiply.bug.fix)

(def multiply *)

; ...
(ns multiply.bug.fix)

(defn multiply 
 "A simple multiplication test"
  [& args] (apply * args))

; ...
(ns multiply.bug.fix)

(defn multiply [& args]
  (reduce * args))

; ...

File: C++/8 kyu - Multiply.cpp
// https://www.codewars.com/kata/multiply/train/cpp
// My solution
int multiply(int a, int b)
{
    return a * b;
}

// ...
auto multiply = [](int a, int b) { return a*b; };

// ...
#include <functional>

std::multiplies<int> multiply;

// ...
template <class T>
inline const T multiply(const T a, const T b)
{
    return a * b;
}

// ...
template <class T>
T multiply(T a,T b){
    return a * b;
}

